src/service.rs

```diff
@@ -59,22 +44,72 @@ pub struct Options {
 //
 #[allow(clippy::type_complexity)]
 #[derive(Debug)]
-pub struct GlobalState<Types: BuilderType> {
+pub struct GlobalState<Types: NodeType> {
+    // identity keys for the builder
+    // May be ideal place as GlobalState interacts with hotshot apis
+    // and then can sign on responsers as desired
+    pub builder_keys: (
+        Types::SignatureKey,                                             // pub key
+        <<Types as NodeType>::SignatureKey as SignatureKey>::PrivateKey, // private key
+    ),
+
+    // data store for the blocks
     pub block_hash_to_block: HashMap<
         BuilderCommitment,
         (
             Types::BlockPayload,
-            <<Types as BuilderType>::BlockPayload as BlockPayload>::Metadata,
-            Arc<JoinHandle<()>>,
-            <<Types as BuilderType>::SignatureKey as SignatureKey>::PureAssembledSignatureType,
-            Types::SignatureKey,
+            <<Types as NodeType>::BlockPayload as BlockPayload>::Metadata,
+            //Arc<JoinHandle<()>>,
```

Comment:
Get rid of the commented out dead code.

Reply:
Resolved the once which is completely unnecessary, kept some minor at 1-2 places to keep some thoughts in mind!